### PR TITLE
Update vSphere CCM to v1.27.0 and CSI to v3.0.2

### PR DESCRIPTION
**Pull request by kron4eg**
**Which issue(s) this PR fixes**:
Xref #2782

**What type of PR is this?**
/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Update vSphere CCM to v1.27.0 and CSI to v3.0.2
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/templates/images/images.go [hunk 1]
```diff
@@ -355,7 +355,9 @@ func optionalResources() map[Resource]map[string]string {
 		// vSphere CCM
 		VsphereCCM: {
 			"1.24.x":    "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.24.2",
-			">= 1.25.0": "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.25.0",
+			"1.25.x":    "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.25.0",
+			"1.26.x":    "gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.26.0",
```
Comment (xmudrii) [inline pkg/templates/images/images.go:359]: As per https://github.com/kubernetes/cloud-provider-vsphere, there are newer patch releases available. Can we please upgrade to those newer patch releases?

File: pkg/templates/images/images.go [hunk 2]
```diff
@@ -365,16 +367,16 @@ func optionalResources() map[Resource]map[string]string {
...
+		VsphereCSIDriver:                    {"*": "gcr.io/cloud-provider-vsphere/csi/release/driver:v3.0.2"},
+		VsphereCSISyncer:                    {"*": "gcr.io/cloud-provider-vsphere/csi/release/syncer:v3.0.2"},
+		VsphereCSIAttacher:                  {"*": "registry.k8s.io/sig-storage/csi-attacher:v4.2.0"},
+		VsphereCSILivenessProbe:             {"*": "registry.k8s.io/sig-storage/livenessprobe:v2.9.0"},
+		VsphereCSINodeDriverRegistar:        {"*": "registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.7.0"},
+		VsphereCSIProvisioner:               {"*": "registry.k8s.io/sig-storage/csi-provisioner:v3.4.0"},
+		VsphereCSIResizer:                   {"*": "registry.k8s.io/sig-storage/csi-resizer:v1.7.0"},
+		VsphereCSISnapshotter:               {"*": "registry.k8s.io/sig-storage/csi-snapshotter:v6.2.1"},
+		VsphereCSISnapshotController:        {"*": "registry.k8s.io/sig-storage/snapshot-controller:v6.2.1"},
+		VsphereCSISnapshotValidationWebhook: {"*": "registry.k8s.io/sig-storage/snapshot-validation-webhook:v6.2.1"},
```
Comment (xmudrii) [inline pkg/templates/images/images.go:379]: Have you verified if there are any changes to snapshotter CRDs?

File: addons/csi-vsphere/vsphere-csi-driver.yaml [hunk 3]
```diff
@@ -160,21 +166,22 @@ roleRef:
...
+  "csi-migration": "true"
   "csi-auth-check": "true"
   "online-volume-extend": "true"
   "trigger-csi-fullsync": "false"
   "async-query-volume": "true"
-  "improved-csi-idempotency": "false"
-  "improved-volume-topology": "false"
   "block-volume-snapshot": "true"
-  "csi-windows-support": "false"
+  "csi-windows-support": "true"
```
Comment (xmudrii) [inline addons/csi-vsphere/vsphere-csi-driver.yaml:175]: Do we need this as we don't support Windows nodes?

**Review by xmudrii — CHANGES_REQUESTED** [review]

File: pkg/templates/images/images.go (hunk 2 above)
Comment (kron4eg) [reply pkg/templates/images/images.go:379]: Yeah, last time they been changed is more then year ago

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**